docs/integration_pipeline_controller.md

```diff
@@ -24,12 +27,17 @@ flowchart TD
   predicate                                   --> clean_environment
   get_resources     --No                      --> error
   get_resources     --Yes                     --> report_status_snapshot
-  report_status_snapshot                    ----> report_status
+  report_status_snapshot                      --> is_snapshot_of_pr_event
+  is_snapshot_of_pr_event            --Yes    --> report_status
+  is_snapshot_of_pr_event            --No     --> is_plr_finished_or_getting_deleted
+  is_plr_finished_or_getting_deleted --Yes    --> remove_finalizer
+  is_plr_finished_or_getting_deleted --No     --> report_status
```

Comment:
I don't understand why there are 3 paths for `Report status if Snapshot was created for Pull requests` in [dragram](https://github.com/dheerajodha/integration-service/blob/STONEINTG-599/docs/integration_pipeline_controller.md), since the right two paths are from `No` of `Is Snapshot created for Pull requests?`, could explain it? Thanks.

Reply:
Those 3 paths explain [this if condition](https://github.com/redhat-appstudio/integration-service/pull/341/files#diff-d2d01b4dc526c804c076232db16ae083ed3381a14d78fc2bbbab8fd4bad59077R106).
* Branch 1: when `gitops.IsSnapshotCreatedByPACPullRequestEvent(snapshot)` returns true.
* Branch 2: when it returns False AND both of the `"h.HasPipelineRunFinished(a.pipelineRun)", "pipelinerunStatus == intgteststat.IntegrationTestStatusDeleted"` function returns False.
* Branch 3: when it returns False AND either of the `"h.HasPipelineRunFinished(a.pipelineRun)", "pipelinerunStatus == intgteststat.IntegrationTestStatusDeleted"` function returns True.

Maybe we can make this part of the diagram shorter in some way...

Reply:
I see your meaning now. Here `Report status if Snapshot was created for Pull requests` means reporting status to GitHub right? But it is not done by integration PLR controller, so I think it should not be displayed here.

Reply:
I agree with you!
Since this change is outside the scope of this PR (as it was not added here), I think it should be tackled within a new PR (if you don't mind).

Reply:
Opened a PR to fix this: https://github.com/redhat-appstudio/integration-service/pull/395